apps/whale-api/src/module.indexer/model/dftx/composite.swap.ts

```diff
@@ -44,33 +40,34 @@ export class CompositeSwapIndexer extends DfTxIndexer<CompositeSwap> {
     const fromAmount: BigNumber = poolSwap.fromAmount
 
     for (const pool of poolIds) {
-      const poolPair = await this.getPoolPair(pool.id)
-      await this.poolSwapIndexer.invalidateSwap(transaction, poolPair.poolPairId, poolSwap.fromTokenId, fromAmount)
-      // fromAmount = ONE.minus(poolPair.commission).times(fromAmount)
+      await this.poolSwapIndexer.invalidateSwap(transaction, `${pool.id}`, poolSwap.fromTokenId, fromAmount)
     }
   }
 
-  async getPoolPair (poolId: number): Promise<PoolPairHistory> {
-    // TODO(fuxingloh): we need to cache this too
-    const poolPair = await this.poolPairHistoryMapper.getLatest(`${poolId}`)
-    if (poolPair === undefined) {
-      throw new IndexerError(`Pool with id ${poolId} not found`)
-    }
-
-    return poolPair
-  }
-
   async getPoolIdsForTokens (compositeSwap: CompositeSwap): Promise<PoolId[]> {
     if (compositeSwap.pools.length > 0) {
       return compositeSwap.pools
     }
 
     const poolSwap = compositeSwap.poolSwap
-    const poolPairToken = await this.poolPairTokenMapper.getPair(poolSwap.fromTokenId, poolSwap.toTokenId)
-    if (poolPairToken === undefined) {
-      throw new IndexerError(`Pool for pair ${poolSwap.fromTokenId}, ${poolSwap.toTokenId} not found`)
+    const fromToken = await this.deFiDCache.getTokenInfo(`${poolSwap.fromTokenId}`)
+    if (fromToken === undefined) {
+      throw new NotFoundApiException('Unable to find fromToken')
+    }
+
+    const toToken = await this.deFiDCache.getTokenInfo(`${poolSwap.toTokenId}`)
+    if (toToken === undefined) {
+      throw new NotFoundApiException('Unable to find toToken')
+    }
+
+    let poolPair = await this.deFiDCache.getPoolPairInfo(`${fromToken.symbol}-${toToken.symbol}`)
+    if (poolPair === undefined) {
+      poolPair = await this.deFiDCache.getPoolPairInfo(`${toToken.symbol}-${fromToken.symbol}`)
```

Comment:
I think this will return pool pair after split instead of before, but probably no better solution now.

Reply:
Hmmm

Reply:
Might be problematic, but I think we can live with this

Reply:
https://github.com/JellyfishSDK/jellyfish/pull/1567 fixes it